Assets/Scripts/UpgradeStore.cs

```diff
@@ -25,10 +40,12 @@ void OnDestroy()
     }
 
 
-    public void OnUpgrade(Upgrade msg)
+    public void OnUpgrade(Upgrade msg, Price msg2)
     {
         amount = msg.upgrade;
         count += amount;
-        print("upgrade count! " + count);
+        price = upgradeCost * System.Math.Pow(multiplier, amount);
+        msg2.price = price;
```

Comment:
After you determine the price, you should send a new event to let everyone know there's a new price.